src/main/java/com/blackducksoftware/integration/hub/global/HubProxyInfo.java

```diff
@@ -18,67 +18,103 @@
  *******************************************************************************/
 package com.blackducksoftware.integration.hub.global;
 
+import java.io.IOException;
 import java.io.Serializable;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class HubProxyInfo implements Serializable {
+import org.apache.commons.lang3.StringUtils;
 
+public class HubProxyInfo implements Serializable {
 	private static final long serialVersionUID = 8813395288911907788L;
 
 	private final String host;
-
-	private final Integer port;
-
-	private final String proxyUsername;
-
-	private final String proxyPassword;
-
+	private final int port;
+	private final String username;
+	private final String password;
 	private final String ignoredProxyHosts;
 
-	private final List<Pattern> ignoredProxyHostPatterns;
-
-	private final boolean hubUrlIgnored;
-
-
-	public HubProxyInfo(final String host, final Integer port, final String noProxyHosts,
-			final List<Pattern> noProxyHostsPatterns, final String proxyUsername, final String proxyPassword,
-			final boolean hubUrlIgnored) {
+	public HubProxyInfo(final String host, final int port, final String username, final String password,
+			final String ignoredProxyHosts) {
 		this.host = host;
 		this.port = port;
-		this.ignoredProxyHosts = noProxyHosts;
-		this.ignoredProxyHostPatterns = noProxyHostsPatterns;
-		this.proxyUsername = proxyUsername;
-		this.proxyPassword = proxyPassword;
-		this.hubUrlIgnored = hubUrlIgnored;
+		this.username = username;
+		this.password = password;
+		this.ignoredProxyHosts = ignoredProxyHosts;
 	}
 
-	public String getHost() {
-		return host;
-	}
+	public URLConnection openConnection(final URL url) throws IOException {
+		if (shouldUseProxyForUrl(url)) {
+			final Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(host, port));
+			setDefaultAuthenticator();
 
-	public Integer getPort() {
-		return port;
-	}
+			return url.openConnection(proxy);
+		}
 
-	public String getProxyUsername() {
-		return proxyUsername;
+		return url.openConnection();
 	}
 
-	public String getProxyPassword() {
-		return proxyPassword;
+	public boolean shouldUseProxyForUrl(final URL url) {
+		final List<Pattern> ignoredProxyHostPatterns = getIgnoredProxyHostPatterns();
+		return !shouldIgnoreHost(url.getHost(), ignoredProxyHostPatterns);
 	}
 
-	public String getIgnoredProxyHosts() {
-		return ignoredProxyHosts;
+	public void setDefaultAuthenticator() {
+		if (username != null && password != null) {
+			Authenticator.setDefault(new Authenticator() {
+				@Override
+				public PasswordAuthentication getPasswordAuthentication() {
+					return new PasswordAuthentication(username, password.toCharArray());
```

Comment:
If we encrypt the password in the constructor then this would need to decrypt the password since it is in use here.